### PR TITLE
Update bundle property to accept module and nomodule

### LINKED ISSUE #211: Add `module` / `nomodule` to the `route.bundle` property
As a developer I want to use the Vaadin.Router components lazy loading feature both in the browsers that support ES modules and in the browsers that do not support it. I expect the `route.bundle` property to offer me a way of specifying both a ES module bundle and a non-ES-module bundle.

DoD:
 - `route.bundle` can be a string, in which case it is loaded as `<script src="..." async>` (the current behavior)
 - `route.bundle` can be an object
```javascript
bundle = {
  module: '/path/to/es-module-bundle.js',
  nomodule: '/path/to/nomodule-bundle.js',
}
```
in which case the bundle from the `nomodule` property is loaded as `<script nomodule src="..." async>` and the bundle from the `module` property is loaded as `<script type="module" src="...">`

### PR DESCRIPTION
**Pull request by web-padawan**
Fixes #211

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-router/218)

### CODE REVIEW

**Comment by vlukashov** [comment]
:lgtm:

---

Reviewed 8 of 8 files at r1.
Review status: :shipit: complete! all files reviewed, all discussions resolved

---



*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-router/218#-:-LGUo9rM87FRWS_CXV2R:bnfp4nl)*

File: docs/vaadin-router/demo/vaadin-router-code-splitting-demos.html [hunk 1]
```diff
@@ -24,8 +24,8 @@ <h3>Lazy Loading JS Bundles</h3>
           ```js
           {
             bundle: {
-              src: '/chunk.js', // required, contains the bundle URL
-              type: 'module' // optional, default to "nomodule"
+              module: '/bundle.esm.js', // for browsers supporting ES modules
+              type: '/bundle.nomodule.js' // fallback for older browsers
```
Comment (moderndeveloperllc) [inline docs/vaadin-router/demo/vaadin-router-code-splitting-demos.html]: Should this be `nomodule` instead of `type`?

Comment (web-padawan) [reply docs/vaadin-router/demo/vaadin-router-code-splitting-demos.html]: Good catch, thanks!

**Comment by moderndeveloperllc** [comment]
@web-padawan How is this going to interact with differential serving? If I have just a couple of simple builds via polymer-cli
```
builds
  |
  |-- esm-bundled
  |-- es5-bundled
```
Should I just not worry about making `bundle` an object and just specify the bundle file name (which should be consistent across builds)?

**Comment by web-padawan** [comment]
@moderndeveloperllc I would rather choose the second option here, just specify the bundle file name. The "module" and "nomodule" is considered as an advanced opt-in feature, which is questionable for developers using `polymer-cli` but might be needed in some other cases.

**Pull request merged**